### DISCUSSION TITLE
Find all payment events for an order

### DISCUSSION THREAD

**Discussion by offlink-io** [swellstores/community#489] [Help] [answered]
Hello :)

I'm currently trying to reproduce what you've set up for order activity:

![image](https://github.com/swellstores/community/assets/128164601/a66e0e26-f04a-4863-bbd3-9c152230070e)

However, I am only able to find the first event linked to a payment:

![image](https://github.com/swellstores/community/assets/128164601/3d6ebb10-d0b9-459a-8dad-dfa98688a02a)

The others are not returned to me even though I had several payments on this order:

![image](https://github.com/swellstores/community/assets/128164601/95c44c81-b910-409b-8153-9d96f0818000)

Is this normal?

Currently, I'm retrieving events as follows for a specific order:

`const events = await swellnode.get(/events?data.id=65b91c12d2c4150011bc0501);
`

Who return:

```json
events: {
  count: 7,
  page_count: 1,
  page: 1,
  results: [
    {
      user_id: '6404938ba7dfa30011e22f3b',
      model: 'orders',
      type: 'order.delivered',
      data: [Object],
      req_id: '65b920854867d70012d83200',
      date_created: '2024-01-30T16:15:02.408Z',
      id: '65b920864867d70012d8321f'
    },
    {
      user_id: '6404938ba7dfa30011e22f3b',
      model: 'orders',
      type: 'order.updated',
      data: [Object],
      req_id: '65b920854867d70012d83200',
      date_created: '2024-01-30T16:15:02.356Z',
      id: '65b920864867d70012d8321c'
    },
    {
      user_id: '6404938ba7dfa30011e22f3b',
      model: 'orders',
      type: 'order.updated',
      data: [Object],
      req_id: '65b920768fa0180012cb973f',
      date_created: '2024-01-30T16:14:46.628Z',
      id: '65b920768fa0180012cb9753'
    },
    {
      user_id: '6404938ba7dfa30011e22f3b',
      model: 'orders',
      type: 'order.updated',
      data: [Object],
      req_id: '65b91e99e155f40012d19b1f',
      date_created: '2024-01-30T16:06:50.248Z',
      id: '65b91e9ae155f40012d19b46'
    },
    {
      user_id: '6404938ba7dfa30011e22f3b',
      type: 'order.paid',
      data: [Object],
      model: 'orders',
      req_id: '65b91e729f7ebe0012771707',
      date_created: '2024-01-30T16:06:10.820Z',
      id: '65b91e729f7ebe0012771728'
    },
    {
      user_id: '6404938ba7dfa30011e22f3b',
      model: 'orders',
      type: 'order.submitted',
      data: [Object],
      req_id: '65b91c12d2c4150011bc04f6',
      date_created: '2024-01-30T15:56:03.056Z',
      id: '65b91c13d2c4150011bc0525'
    },
    {
      user_id: '6404938ba7dfa30011e22f3b',
      model: 'orders',
      type: 'order.created',
      data: [Object],
      req_id: '65b91c12d2c4150011bc04f6',
      date_created: '2024-01-30T15:56:02.961Z',
      id: '65b91c12d2c4150011bc051c'
    }
  ]
}
```

As you can see, I only have one `order.paid` event.

I may have misunderstood how this works, but aren't I supposed to have 3 in this case?

Thx for your help

**Comment by ericingram** [comment, marked as answer]
@la-ruee-vers-le-bois the activity feed in the dashboard is made up of events from multiple models. Payments to orders are recorded in the event type `payment.succeeded` and `payment.failed`. The order `paid` event is only triggered when the order is fully paid for the first time, while any additional payments/refunds won't re-trigger the event because it's common to make adjustments to orders over a short period of time.

We hope to introduce an API for the activity feed itself in the future, at the same time expanding it across other areas of the product, but we have no active plans to implement that just yet.

**Reply by offlink-io** [reply]
Thx @ericingram for your quick response.

Finally, using the content of order.payments I manage to get the result I want.

Thx again :)
